File: csharp/src/impl/DefaultPersistentComparator.cs
namespace Volante.Impl
{
    using System;
    using Volante;

    public class DefaultPersistentComparator<K, V> : PersistentComparator<K, V> where V : class,IPersistent, IComparable<V>, IComparable<K>
    {
        public override int CompareMembers(V m1, V m2)
        {
            return m1.CompareTo(m2);
        }

        public override int CompareMemberWithKey(V mbr, K key)
        {
            return mbr.CompareTo(key);
        }
    }
}

File: java/src15/org/nachodb/impl/DefaultPersistentComparator.java
package org.nachodb.impl;

import org.nachodb.*;

public class DefaultPersistentComparator<T extends IPersistent&Comparable> extends PersistentComparator<T> { 
    public int compareMembers(T m1, T m2) {
        return m1.compareTo(m2);
    }
        
    public int compareMemberWithKey(T mbr, Object key) { 
        return mbr.compareTo(key);
    }
}